CMakeLists.txt

```diff
@@ -158,12 +158,11 @@ message(STATUS
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 if(JPEGXL_STATIC)
-  set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
```

Comment:
I just realized that we may still want this in the `if (NOT MSVC AND NOT APPLE)` branch. Let’s perhaps move it there instead of removing it?

Reply:
I'm happy to move it back if that's what's needed I just haven't seen a case where someone overrides that value in practice.

Its typically done in the Platform files like the Windows-GNU one you pointed at had it set like this https://gitlab.kitware.com/cmake/cmake/-/blob/master/Modules/Platform/Windows-GNU.cmake#L27

Reply:
From what I remember, the rationale is that when building statically, we want to find the static (`.a`) version of libraries specifically and not the dynamic (`.dll.a`) one.

Reply:
Updated with fix.

Reply:
Do we have static builds in CI actions that can fail or succeed depending on getting this right?

I dunno why we had that line in the first place, it indeed seems like something that should not need to be manually set (cmake should know best how to do static or dynamic library search on each platform) but maybe there was a need somehow.

I am used to shared libraries being `.so` and static ones being `.a`, but I don't know how it is on non-linux platforms.

Reply:
Maybe we did it like that because on linux, we don't do a 'real' static build, and still depend on stdlib.